src/main/groovy/org/spongepowered/gradle/ore/OreDeployTask.groovy

```diff
@@ -0,0 +1,156 @@
+    String apiKey
+
+    Configuration deploy = project.configurations.archives
+
+    @TaskAction
+    void run() {
+        Logger logger = project.logger
+        def artifacts = deploy.allArtifacts
+        PublishArtifact plugin = artifacts.find { a -> (a.type == "jar") }
+        PublishArtifact sig = artifacts.find { a -> (a instanceof Signature) }
+        if (plugin == null) {
+            throw new InvalidUserDataException("Plugin file not found.")
+        }
+        if (sig == null) {
+            throw new InvalidUserDataException("Signature file not found.")
+        }
+
+        logger.quiet('Publishing ' + plugin.name + ' to ' + instanceUrl + '.')
+        logger.quiet('  Recommended: ' + recommended)
+        logger.quiet('  Channel: ' + channel)
+
+        if (instanceUrl.endsWith("/")) {
+            instanceUrl = instanceUrl.substring(0, instanceUrl.length() - 1)
+        }
+        URL projectUrl
+        try {
+            String pluginId = project.sponge.plugin.id
+            String str = instanceUrl + '/api/projects/' + pluginId + '/versions/' + project.version
+            projectUrl = new URL(str)
+        } catch (MalformedURLException e) {
+            throw new InvalidUserDataException("Invalid project URL", e)
+        }
+
+        if (apiKey == null) {
+            apiKey = project.property('oreDeploy.apiKey')
+        }
+        if (channel == null) {
+            throw new InvalidUserDataException("Missing channel name.")
+        }
+        HttpEntity requestEntity = MultipartEntityBuilder.create()
+            .addPart('apiKey', new StringBody(apiKey))
+            .addPart('channel', new StringBody(channel))
+            .addPart('recommended', new StringBody(recommended as String))
+            .addPart('pluginFile', new FileBody(plugin.file))
+            .addPart('pluginSig', new FileBody(sig.file))
+            .build()
+
+        HttpURLConnection conn = null
+        InputStream is = null
+        try {
+            conn = (HttpURLConnection) projectUrl.openConnection()
+            conn.doOutput = true
+            conn.requestMethod = 'POST'
+            conn.useCaches = false
+            conn.readTimeout = 10000
+            conn.connectTimeout = 10000
+            conn.setRequestProperty('Connection', 'Keep-Alive')
+            conn.setRequestProperty('Cache-Control', 'no-cache')
+            conn.setRequestProperty('Content-Length', requestEntity.contentLength as String)
+            conn.setRequestProperty('Content-Type', requestEntity.contentType.value)
+
+            conn.outputStream.withStream { requestEntity.writeTo(it) }
+
+            conn.connect()
+            if (conn.responseCode != HttpURLConnection.HTTP_CREATED) {
+                logger.error('[failure] ' + conn.responseCode + ' ' + conn.responseMessage)
+                is = conn.errorStream
+                printErrorStream is
+                is.close()
+                is = null
+                throw new GradleException()
```

Comment:
You should also use `conn.errorStream.withStream` here. Also, instead of printing the error to the logger and then throwing the exception you should include the error message in the exception. Otherwise, Gradle will display an error with `(no error message)`